### PR TITLE
Refactor CMake options to default value OFF

### LINKED ISSUE #259: Refactor CMake options to default value OFF
**TL;TR**: *If an option or preprocessor definition is not specified, it is always `OFF`.*

## Motivation

While working on the new Sphinx-based docs and fiddling with documenting CMake options and cache entries (see https://github.com/lexicalunit/nanodbc/pull/258#discussion_r111134321) I realised, nanodbc-specific options:

* Mix On and OFF as default values
* Do not follow [CMake OPTION](https://cmake.org/cmake/help/latest/command/option.html) semantics:
    > Provide an option for the user to select as `ON` or `OFF`. If no initial value is provided, `OFF` is used.

## Proposal

This task proposes to refactor all `NANODBC_*` options and corresponding preprocessor definitions to follow the CMake `OPTION` semantics with default value `OFF`

## Changes

Naming options to follow the CMake option semantics with "none", `USE` or `HANDLE` wording is tricky, so it is better to rationalise names using `DISABLE` or `ENABLE`. 

**opt-out** options, with default `OFF`:

* `NANODBC_DISABLE_ASYNC`
* `NANODBC_ENABLE_LIBCXX` to `NANODBC_DISABLE_LIBCXX`
* `NANODBC_EXAMPLES` to `NANODBC_DISABLE_EXAMPLE` (`example` target after folder name)
* `NANODBC_INSTALL` to `NANODBC_DISABLE_INSTALL`
* `NANODBC_TEST` to `NANODBC_DISABLE_TEST` (`test` target after folder name)

**opt-in** options, with default `OFF`:

* `NANODBC_USE_UNICODE` to `NANODBC_ENABLE_UNICODE`
* `NANODBC_USE_BOOST_CONVERT` to `NANODBC_ENABLE_BOOST_CONVERT` (or  `NANODBC_ENABLE_BOOST_LOCALE` after Boost.Locale lib name)
* `NANODBC_HANDLE_NODATA_BUG` no need to rename, alternatively, rename to `NANODBC_ENABLE_NODATA_FIX`

String cache entry:

* `NANODBC_ODBC_VERSION` no need to rename.

------

Feedback is welcome!

### PR DESCRIPTION
**Pull request by mloskot**
Follow CMake convention for `OPTION` semantic:

> if an option or preprocessor definition is not specified, it is always `OFF`.

Closes #259

### CODE REVIEW

**Comment by mloskot** [comment]
@lexicalunit How you feel about one?

File: CMakeLists.txt [hunk 1]
```diff
@@ -203,7 +206,7 @@ endif()
 
 add_library(nanodbc nanodbc/nanodbc.cpp nanodbc/nanodbc.h)
 
-# TODO: Consider version/soverson support
+# TODO: Consider version/soversin support
```
Comment (lexicalunit) [inline CMakeLists.txt]: What's soversin or soverson?

**Comment by lexicalunit** [comment]
LGTM with tests passing too 🎉

**Comment by lexicalunit** [comment]
Ah, I guess we have to resolve conflicts first.

File: CMakeLists.txt (hunk 1 above)
Comment (mloskot) [inline CMakeLists.txt]: Good catch. Keyboard failure ;-)

**Comment by mloskot** [comment]
Updated the PR resolving the conflicts. I'll merge as soon as the builds have completed.

**Comment by mloskot** [comment]
@lexicalunit Thanks for review.

**Pull request merged**